### ISSUE TITLE
chore: Get rid of `trace_many` dependency in bad token detection

### ISSUE THREAD

**Issue by MartinquaXD** [cowprotocol/services#2412]
### Background
Currently our bad token detection requires a node that supports `trace_many` which for example rules out `geth`. It should be possible to implement a similar logic using helper solidity contracts we use with state overrides.
This would basically trade in the dependency on `trace_many` with state overrides. But we already have a very big dependency on that so it would be okay. Another downside would be the increased complexity (in terms of implementation and execution).
And finally our goal is to get rid of bad token detection soon-ish anyway so it might not make sense to spend time on the old stuff.

Some of the techniques that might come in handy have been used and described in https://github.com/cowprotocol/services/pull/1575.

Basically this issue is just a braindump for the idea so we can track it if we actually want to do it.

### Acceptance criteria
Bad token detection no longer depends on `trace_many`.

**Comment by MartinquaXD** [comment]
We are considering to get rid of bad token detection completely in the near future so I think it no longer makes sense to tackle this issue.
